src/plugins/ucx/ucx_backend.cpp

```diff
@@ -385,11 +384,15 @@ void nixlUcxEngine::progressFunc()
             if (made_progress && !wid)
                 notifProgress();
         }
+        timeout = false;
 
-        while (poll(pollFds.data(), pollFds.size(), -1) <= 0)
+        int ret;
+        while ((ret = poll(pollFds.data(), pollFds.size(), pthrDelay.count())) < 0)
             NIXL_ERROR << "Call to poll() was interrupted, retrying. Error: " << strerror(errno);
```

Comment:
what kind of errors happens here? do we really need errors in log for any type of error? maybe we could silently handle EAGAIN, EINTR, etc or convert to NIXL_DEBUG? How about something like this:
```suggestion
        while (true) {
            int ret = poll(pollFds.data(), pollFds.size(), pthrDelay.count());
            if (ret < 0) {
                if (errno != EAGAIN) && (errno != EINTR)) {
                    NIXL_ERROR << "poll() failed with error: " << strerror(errno) << ", retrying.";
                }
            } else {
                timeout = (ret > 0);
                break;
            }
        }
```

Reply:
regarding logs i agree with @evgeny-leksikov, but afair it was @yosefe comment to log it, but my impression it was debug or trace before..
maybe it is indeed makes sence no to log for EAGAIN and EINTR to avoid

Reply:
Hmm, I can see why we would want to avoid spamming this on "error" level, but being able to determine by the log that progress thread is stuck in infinite EAGAIN loop also has some value IMO.

Reply:
can we log it then with DEBUG or TRACE rather than ERRROR?